core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -488,12 +508,26 @@ default void finishStatisticsCollection(ConnectorSession session, ConnectorTable
 
     /**
      * Begin the atomic creation of a table with data.
+     *
+     * @deprecated Use {@link #beginCreateTable(ConnectorSession, ConnectorTableMetadata, Optional, RetryMode)}
      */
+    @Deprecated
     default ConnectorOutputTableHandle beginCreateTable(ConnectorSession session, ConnectorTableMetadata tableMetadata, Optional<ConnectorNewTableLayout> layout)
     {
         throw new TrinoException(NOT_SUPPORTED, "This connector does not support creating tables with data");
     }
 
+    /**
+     * Begin the atomic creation of a table with data.
+     */
+    default ConnectorOutputTableHandle beginCreateTable(ConnectorSession session, ConnectorTableMetadata tableMetadata, Optional<ConnectorNewTableLayout> layout, RetryMode retryMode)
+    {
+        if (retryMode != RetryMode.NO_RETRIES) {
+            throw new TrinoException(NOT_SUPPORTED, "This connector does not support query retries");
```

Comment:
`query retries` implies full query retires, while it is more of the failure recovery in general being a concern. How about changing the message to `This connector does not support creating tables with failure recovery enabled`?

Reply:
I would keep it as is unless we also change the configuration option `retry-policy` and matching session property to not refer retries explicitly. I think it is more understandable if we keep language consistent.